WordPress/Classes/ViewRelated/Post/WPPostViewController.m

```diff
@@ -2051,7 +2051,26 @@ - (void)editorDidPressSettings:(WPEditorViewController *)editorController
 
 - (void)editorDidPressMedia:(WPEditorViewController *)editorController
 {
-    [self showMediaPicker];
+    if (self.post.blog.canUploadFiles) {
+        [self showMediaPicker];
+        return;
+    }
+
+    // first button of the toolbar
+    UIBarButtonItem *mediaButton = [editorController.toolbarView.items objectAtIndex:0];
+    UIView *buttonView = mediaButton.customView;
+    CGRect buttonFrame = [self.view convertRect:buttonView.frame
+                                       fromView:editorController.toolbarView];
+    CGRect targetFrame = CGRectMake(buttonFrame.origin.x,
+                                    buttonFrame.origin.y,
+                                    buttonFrame.size.width,
+                                    0.0);
+    NSString *tooltipText = NSLocalizedString(@"You are not allowed to add media on this blog",
```

Comment:
I actually went exactly through the same thought process. I started with "Contributors can't add images to their posts" and then because the feature is  based on a set - definite - capability property, I went for the "you're not allowed" version. I wanted the message to be very explicit. By saying that it makes it clear to the user that they have to ask for permissions to an administrator if they want to unblock this.
Maybe we could go with something similar but less harsh. How about "you haven't been granted permission to attach media on this blog" or something like this? This implies that you ~could~ get granted permission...